docs/did-spec/spec.md

```diff
@@ -0,0 +1,343 @@
+
+### Resolve a DID
+
+Resolution of a KILT DID is opaque over the type of DID being resolved.
+There are two ways that KILT DIDs can be resolved: using the KILT SDK and using the DIF Universal Resolver.
+
+In the first case, the KILT SDK exposes functions to resolve KILT DIDs referring to whole DID Documents, just a verification key by its ID, or just a service endpoint by its ID or its type.
+For more details about the functions and types provided by the KILT SDK, please visit the [official SDK documentation][kilt-did-docs].
+
+KILT also provides a resolution driver for the [DIF Universal Resolver][dif-universal-resolver].
+In this case, the DID resolution process and result follow the [W3C DID Core Specification][did-core-spec] concerning the structure of the DID Document and the resolution metadata.
+
+Specifically for resolution metadata, every DID resolution result contains document resolution metadata depending on the state of the DID being resolved:
+
+<table>
+<tr>
+<td> DID status </td> <td> Metadata structure </td>
+</tr>
+<tr>
+<td> Deleted DID </td>
+<td>
+
+```json
+{
+    "deactivated": true
+}
+```
+
+</td>
+</tr>
+<tr>
+<td> Migrated light DID with the full DID not deactivated </td>
+<td>
+
+```json
+{
+    "canonicalId": "did:kilt:<did_identifier>",
+    "deactivated": false
+}
+```
+
+</td>
+</tr>
+<tr>
+<td> Un-migrated light DID or a full DID </td>
+<td>
+
+```json
+{
+    "deactivated": false
+}
+```
+
+</td>
+</tr>
+</table>
+
+If the first case where `deactivated` is true, the resulting `didDocument` will be a minimal document containing only the `id` and `@context` properties as specified in the DID Core specification.
+
+On the KILT blockchain, information related to a full DID is stored under multiple storage maps within the `did` pallet, specifically:
+
+- `did -> did(AccountId32): Option<DidDetails>`: maps from a DID identifier to its details, if present.
+An instance of `DidDetails` contains all the keys under the control of the DID subject.
+- `did -> didBlacklist(AccountId32): Option<()>`: maps from a DID identifier to an optional empty tuple indicating whether a given DID has been deleted (optional tuple is not null) or not (optional tuple is null).
+- `did -> serviceEndpoints(AccountId32, Vec<u8>): Option<DidEndpoint>`: maps from the concatenation of hashed DID identifier and hashed service ID to the service endpoints details, if present.
+Upon lookups, by only providing the first key, it is possible to retrieve all the service IDs under a given DID identifier.
+An instance of `DidEndpoint` contains the service details, i.e., service ID, a set of service types, and a set of service URLs.
+
+> The validity and authenticity of the information stored on the blockchain are guaranteed by the same blockchain, which verifies that all DID management operations other than creation are signed by the DID authentication key, as described in the section relative to each operation.
```

Comment:
> as described in the section relative to each operation.

What do you mean here? Respective sections in the docs?

Reply:
Yeah I was referring to the sections in the docs that describe those other operations, for instance DID update or DID deletion.

Reply:
Can we say the KILT blockchain?